core/liquidity/v2/sla.go

```diff
@@ -0,0 +1,229 @@
+package liquidity
+
+import (
+	"encoding/binary"
+	"math/rand"
+	"time"
+
+	"code.vegaprotocol.io/vega/libs/crypto"
+	"code.vegaprotocol.io/vega/libs/num"
+)
+
+// TODO karel - real Tendermint txs should be used.
+type TX struct {
+	ID string
+}
+
+func (t TX) Hash() []byte {
+	return crypto.Hash([]byte(t.ID))
+}
+
+// ResetSLAEpoch should be called at the beginning of epoch to reset per epoch performance calculations.
+func (e *Engine) ResetSLAEpoch(now time.Time) {
+	for party, commitment := range e.slaPerformance {
+		if e.doesLPMeetsCommitment(party) {
+			commitment.start = now
+		}
+
+		commitment.s = 0
+	}
+
+	e.slaEpochStart = now
+}
+
+func (e *Engine) BeginBlock(txs []TX) {
```

Comment:
Not blocking this PR because I know you've not wired it in yet but: Can we currently get all the transaction for a block upfront to be able to do this? Wondering if we can just hash the block time instead, surely no one can guess that before it happens.